src/magic_folder/test/test_snapshot.py

```diff
@@ -255,3 +274,119 @@ def test_snapshots_with_parents(self, content1, content2, filename):
             d,
             succeeded(Always()),
         )
+
+    @given(
+        content1=binary(min_size=1),
+        content2=binary(min_size=1),
+        filename=magic_folder_filenames(),
+    )
+    def test_serialize_deserialize_snapshot(self, content1, content2, filename):
+        """
+        create a new snapshot (this will have no parent snapshots).
+        """
+        data1 = io.BytesIO(content1)
+
+        snapshots = []
+        d = create_snapshot(
+            name=filename,
+            author=self.alice,
+            data_producer=data1,
+            snapshot_stash_dir=self.stash_dir,
+            parents=[],
+        )
+        d.addCallback(snapshots.append)
+
+        self.assertThat(
+            d,
+            succeeded(Always()),
+        )
+
+        # now modify the same file and create a new local snapshot
+        data2 = io.BytesIO(content2)
+        d = create_snapshot(
+            name=filename,
+            author=self.alice,
+            data_producer=data2,
+            snapshot_stash_dir=self.stash_dir,
+            parents=snapshots,
+        )
```

Comment:
Might be better / more-explicit as `parents=[snapshots[0]]` to make a new list ... but also *maybe*  `create_snapshot` should take care to deep-copy the incoming parents list?

Reply:
I missed this out in the earlier commits. Fixed now.